grpc/proto/v1/base.proto

```diff
@@ -54,3 +54,81 @@ message ObjectProperties {
   ObjectPropertiesValue value = 1;
   string prop_name = 2;
 }
+
+
+message TextArray {
+  repeated string values = 1;
+}
+
+message IntArray {
+  repeated int64 values = 1;
+}
+
+message NumberArray {
+  repeated double values = 1;
+}
+
+message BooleanArray {
+  repeated bool values = 1;
+}
+
+message Filters {
+  enum Operator {
+    OPERATOR_UNSPECIFIED = 0;
+    OPERATOR_EQUAL = 1;
+    OPERATOR_NOT_EQUAL = 2;
+    OPERATOR_GREATER_THAN = 3;
+    OPERATOR_GREATER_THAN_EQUAL = 4;
+    OPERATOR_LESS_THAN = 5;
+    OPERATOR_LESS_THAN_EQUAL = 6;
+    OPERATOR_AND = 7;
+    OPERATOR_OR = 8;
+    OPERATOR_WITHIN_GEO_RANGE = 9;
+    OPERATOR_LIKE = 10;
+    OPERATOR_IS_NULL = 11;
+    OPERATOR_CONTAINS_ANY = 12;
+    OPERATOR_CONTAINS_ALL = 13;
+  }
+
+  Operator operator = 1;
+  // protolint:disable:next REPEATED_FIELD_NAMES_PLURALIZED
+  repeated string on = 2  [deprecated = true];  // will be removed in the future, use path
+  repeated Filters filters = 3;
+  oneof test_value{
+    string value_text = 4;
+    int64 value_int = 5;
+    bool value_boolean = 6;
+    double value_number = 7;
+    TextArray value_text_array = 9;
+    IntArray value_int_array = 10;
+    BooleanArray value_boolean_array = 11;
+    NumberArray value_number_array = 12;
+    GeoCoordinatesFilter value_geo = 13;
+  };
+  FilterTarget target = 20; // leave space for more filter values
+}
+
+message FilterReferenceSingleTarget {
+  string on = 1;
+  FilterTarget target = 2;
+}
+
+message FilterReferenceMultiTarget {
+  string on = 1;
+  FilterTarget target = 2;
+  string string_target_collection = 3;
```

Comment:
Is the `string_` prefix required here since the type is defined as a string already?

Reply:
yeah, not needed. thx :)